cats/api_query.py

```diff
@@ -12,9 +12,19 @@ def get_tuple(postcode) -> list[tuple[str, int]]:
     # just get the first part of the postcode
     postcode = postcode.split()[0]
     
+    # get the time (as a datetime object) for the top of
+    # the current hour or half hour in UTZ
+    dt = datetime.now(timezone.utc)
+    if dt.minute > 30:
+        dt = dt.replace(minute=31, second=0, microsecond=0)
+    else:
+        dt = dt.replace(minute=1, second=0, microsecond=0)
```

Comment:
I'm sorry @andreww I'm still stuck at this! ;) It's now 19:35UTC and a request to the API gives me a forecast with the first timeslot being 19:30 to 20:00. Which -- I think -- is what I want. Replacing the time with 19:31 would lead to exactly the same data, no? 

Same on the other side of the half-hour - if making a requests at 17:16, i'd expect the forecast data to start with `{'from': '17:00', 'to': '17:30'}`.. same for 17:01. What am I missing here?

Reply:
Ah - I guess the `timestamp` must be the same string in the request url for the cache to be hit!

Reply:
Exactly - this all assumes the same request URL will always give us the same data (it should, if it's RESTfull) so the code turns the time into a string that represents the half hour where we want the data set to start from. Then it is just a case of asking for the same URL each time we need it within the same half hour window. The other thing to know is that if we were to ask for data starting at 17:00 (not 17:01) first period in the forecast data returned is for 16:30 - 17:00.  Do you think this reasoning needs to go in a comment?

Reply:
All clear now, thank you for clarifying. Yes I think a comment is needed. I would say the assumed knowledge for understanding that bit of code is:

 - That the API yields a list of half-hour time slots from the hour and the half-hour
 - The caching works by making a request to the exact same url (makes sense in retrospect but maybe not completely obvious before hand for people with limited exposure to web services.)
 - Asking for data at HH:00 or HH:30 gives you a forecast starting half and hour too early compared to HH:01 or HH:31.

Reply:
I think this is ready for merging.

Reply:
indeed! Thanks for adding the docstrings and comments.